homeassistant/components/wemo/__init__.py

```diff
@@ -79,13 +81,38 @@ def coerce_host_port(value: str) -> HostPortTuple:
 )
 
 
+@dataclass
+class WemoData:
+    """Component state data."""
+
+    discovery_enabled: bool
+    static_config: Sequence[HostPortTuple]
+    registry: pywemo.SubscriptionRegistry
+    config_entry_data: WemoConfigEntryData | None = None
```

Comment:
It's OK to type like this to avoid asserting the type everywhere:
```suggestion
    # config_entry_data is set when the config entry is loaded and unset when it's
    # unloaded. It's a programmer error if config_entry_data is accessed when the
    # config entry is not loaded
    config_entry_data: WemoConfigEntryData = None  # type: ignore[assignment]
```

Reply:
Done in [cfde9bb](https://github.com/home-assistant/core/pull/96620/commits/cfde9bbe83ce428eed16a5bf1c590c4c764c8a02)